src/Query/TimeoutExecutor.php

```diff
@@ -11,20 +12,34 @@ final class TimeoutExecutor implements ExecutorInterface
     private $executor;
     private $loop;
     private $timeout;
+    private $config;
 
     public function __construct(ExecutorInterface $executor, $timeout, LoopInterface $loop = null)
     {
         $this->executor = $executor;
         $this->loop = $loop ?: Loop::get();
         $this->timeout = $timeout;
+        $this->config = \React\Dns\Config\Config::loadSystemConfigBlocking();
     }
 
     public function query(Query $query)
     {
-        return Timer\timeout($this->executor->query($query), $this->timeout, $this->loop)->then(null, function ($e) use ($query) {
+        return $this->tryQuery($query, $query->name);
+    }
+    public function tryQuery(Query $query,$queryName="", $index=0){
```

Comment:
I think we don't need the new `tryQuery()` method here, maybe we should add this logic to the `query()` function instead. WDYT?

Reply:
Because I want keep code style like `dns\src\Query\RetryExecutor.php` file, And function `tryQuery` has recursive call,use a Independent function will be better

Reply:
Oh, didn't notice :D